File: ehr-dashboard/frontend/src/types/ehr.ts
export interface LastSync {
  status: string;
  started_at: string;
  completed_at: string | null;
  patients_processed: number;
  conditions_processed: number;
  medications_processed: number;
  error_message: string | null;
}

export interface EhrSource {
  code: string;
  name: string;
  enabled: boolean;
  last_sync: LastSync | null;
}

export interface SyncResult {
  source: string;
  status: string;
  patients_processed: number;
  conditions_processed: number;
  medications_processed: number;
}

File: ehr-dashboard/backend/app/schemas/ehr.py
from datetime import datetime

from pydantic import BaseModel


class LastSyncResponse(BaseModel):
    status: str
    started_at: datetime
    completed_at: datetime | None
    patients_processed: int
    conditions_processed: int
    medications_processed: int
    error_message: str | None


class EHRSourceResponse(BaseModel):
    code: str
    name: str
    enabled: bool
    last_sync: LastSyncResponse | None = None
